Add delay in emulated_hue after PUT

grogi opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

An Echo device changes the state of the device by submitting a PUT to `/api/{username}/lights/{entity_id}/state`. Immediately after the call, it executes a GET to `/api/{username}/lights/{entity_id}` to validate if the change was successful. 

In many cases the state of the device will not be updated to be reflected in the subsequent GET command. When there is a discrepancy between the state submited and received, Echo would complain with  '/.../ appears to be malfunctioning' announcement. This change introduces a short artificial delay of 250ms before the PUT method returns to allow for the change to successfully propagate, which keeps the Echo device pleased. 

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

emulated_hue:
   listen_port: 80
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #31438 
- This PR is related to issue: #29899
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

grogi committed:
Delay after PUT to improve Echo compatibility. 

Added artificial delay of 100ms after a PUT to allow the changes to propagate. 
This prevents the '/.../ appears to be malfunctioning' message from Echo devices.

grogi committed:
Update hue_api.py

Fixed the comments.

joogps reviewed (changes requested):
I saw this is your first pull request. Welcome to the community!

Did you format your code with black? The tests are failing and pointing to that. It should happen automatically when you commit, maybe you committed using the `-n` flag to skip tests?

joogps commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -555,6 +556,10 @@ async def put(self, request, username, entity_number):
                     create_hue_success_response(entity_number, val, parsed[key])
                 )
 
+        # Echo fetchs the state immediately after the PUT method returns.
```
I think you misspelled "fetches" here

joogps commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -555,6 +556,10 @@ async def put(self, request, username, entity_number):
                 )
 
+        # Echo fetchs the state immediately after the PUT method returns.
+        # Waiting here a short while to allow for the changes to propagate.
```
```suggestion
        # Waiting here for a short time allows the changes to propagate.
```

grogi commented:
> I saw this is your first pull request. Welcome to the community!
> 
> Did you format your code with black? The tests are failing and pointing to that. It should happen automatically when you commit, maybe you committed using the `-n` flag to skip tests?

No, I am properly setting up the environment right now.

joogps commented:
If you need any help...

grogi committed:
emulated_hue: code cleanup and formating

Formatted code with black
The duplicate config resource generation was moved to a function.

grogi committed:
Fixed flake8 complained.

joogps commented:
I tested these changes in my setup and it didn't seem to change much... The "device seems to be malfunctioning" error is still reported. Did it stop happening to you?

grogi commented:
I was consistently getting this message before. After introducing 100ms delay it was significantly reduced, but one wall switch would still trigger the issue in approximately 50% of cases. Increasing it to 250ms seems to completely fix the problem.

joogps commented:
There are multiple ongoing discussions here about this issue, and I hope this one could help fix it.

Since I don't have writing access to this repo, my approval of these changes is merely symbolic. You'll need approval from a maintainer for merging.
